src/nccl_ofi_net.c

```diff
@@ -1182,13 +1183,16 @@ int nccl_net_ofi_info_properties(struct fi_info *nic_prov, int dev_id, int num_d
 	/* name is NULL if device is a part of multirail config */
 	/* overriding default name only if value is available from provider */
 	if (nic_info->device_attr->name) {
-		dev_props.name = strdup(nic_info->device_attr->name);
+		if (dev_props->name) {
```

Comment:
there's no need for this.  This is a programming bug.  If you want to assert == NULL, that's fine.